src/Elastic.Apm/Report/PayloadSender.cs

```diff
@@ -33,7 +34,7 @@ internal class PayloadSender : IDisposable, IPayloadSender
 
 		static PayloadSender() => ServicePointManager.DnsRefreshTimeout = DnsTimeout;
 
-		internal PayloadSender(AbstractLogger logger, IConfigurationReader configurationReader)
+		internal PayloadSender(AbstractLogger logger, IConfigurationReader configurationReader, HttpMessageHandler handler)
```

Comment:
small nit: could we make the `handler` optional with `= null`? That way in `AgentComponents.cs` we don't have to pass `null`. This is the approach we usually used previously in the code-base.

Reply:
sure, will amend